yspryn/hw9/fraction.py

```diff
@@ -0,0 +1,132 @@
+            return Fraction(value1.num*value2.denom, value1.denom*value2.num)
+
+    def __rtruediv__(self, other):
+        value1 = Fraction.convert(other)
+        value2 = Fraction.convert(self)
+        if (value1.num * value2.denom) % (value1.denom * value2.num) == 0:
+            return (value1.num * value2.denom) / (value1.denom * value2.num)
+        else:
+            return Fraction(value1.num * value2.denom, value1.denom * value2.num)
+
+    # add
+    def __add__(self, other):
+        value1 = Fraction.convert(self)
+        value2 = Fraction.convert(other)
+        denominator = value1.denom * value2.denom
+        numerator = value1.num * value2.denom + value2.num * value1.denom
+        if numerator % denominator == 0:
+            return numerator / denominator
+        return Fraction(numerator, denominator)
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    # subtraction
+    def __sub__(self, other):
+        value1 = Fraction.convert(self)
+        value2 = Fraction.convert(other)
+        denominator = value1.denom * value2.denom
+        numerator = value1.num * value2.denom - value2.num * value1.denom
+        if numerator % denominator == 0:
+            return numerator / denominator
+        return Fraction(numerator, denominator)
+
+    def __rsub__(self, other):
+        value1 = Fraction.convert(other)
+        value2 = Fraction.convert(self)
+        denominator = value1.denom * value2.denom
+        numerator = value1.num * value2.denom - value2.num * value1.denom
+        if numerator % denominator == 0:
+            return numerator / denominator
+        return Fraction(numerator, denominator)
+
+    # multiplication
+    def __mul__(self, other):
+        value1 = Fraction.convert(self)
+        value2 = Fraction.convert(other)
+        denominator = value1.denom * value2.denom
+        numerator = value1.num * value2.num
+        if numerator % denominator == 0:
+            return numerator / denominator
+        return Fraction(numerator, denominator)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __bool__(self):
+        if self.denom and self.num:
+            return True
+        else:
+            return False
+
+    # >=
+    def __ge__(self, other):
+        return self.num/self.denom >= other.num/other.denom
+
+    # >
+    def __gt__(self, other):
+        return self.num / self.denom > other.num / other.denom
+
+    # <
+    def __lt__(self, other):
+        return self.num / self.denom < other.num / other.denom
+
+    # <=
+    def __le__(self, other):
+        return self.num / self.denom <= other.num / other.denom
+
+    def __eq__(self, other):
+        if self.denom == other.denom and self.num == other.num:
+            return True
+        else:
+            return False
+
+    # absolute value
+    def __abs__(self):
+        if self.denom < 0:
+            self.denom *= -1
+        elif self.num < 0:
+            self.num *= -1
```

Comment:
and if both < 0

Reply:
here you modify self object, but you should not do that

return Fraction(abs(self.num), self.denom)  # should be enough